### ISSUE TITLE
jquery wants mutable document.documentElement to test for nonstandard getElementByID

### ISSUE THREAD

**Issue by kpreid** [google/caja#1448]
Original [issue 1448](https://code.google.com/p/google-caja/issues/detail?id=1448) created by felix8a on 2012-03-01T09:19:51.000Z:

jquery-1.7.1 creates an element &lt;a name='foo'&gt;
and then adds it with
  document.documentElement.insertBefore(foo, null)
in order to check if getElementByID('foo') will return that name='foo' element
(which happens in IE&lt;=7).

If getElementByID has that broken behavior, jquery modifies a few of its functions
to reliably reject the false hits.

In caja, the insertBefore complains to console,
because documentElement is not mutable,
but silently proceeds (without inserting the node).

Basically, this is harmless except in IE&lt;=7, where jquery will falsely believe that
getElementByID is standards-compliant.

Options:
1. modify domado so that documentElement is mutable.  This should be safe to do, but probably requires adding another layer of &lt;div&gt;.
2. modify domado so that getElementByID is standards-compliant on IE&lt;=7.
3. modify jquery to use a different test that will work reliably in a Caja environment.
4. ignore the issue since IE&lt;=7 is not well supported by Caja anyway.
